python/tank/authentication/login_dialog.py

```diff
@@ -123,14 +128,30 @@ def __init__(self, is_session_renewal, hostname=None, login=None, fixed_host=Fal
         self.ui.forgot_password_link.linkActivated.connect(self._link_activated)
 
         self.ui.site.editingFinished.connect(self._strip_whitespaces)
+        self.ui.site.textEdited.connect(self._site_url_changed)
         self.ui.login.editingFinished.connect(self._strip_whitespaces)
         self.ui._2fa_code.editingFinished.connect(self._strip_whitespaces)
         self.ui.backup_code.editingFinished.connect(self._strip_whitespaces)
 
-        url = self.ui.site.text().encode("utf-8")
-        if is_sso_enabled_on_site(url):
+        self._url_to_test = self.ui.site.text().encode("utf-8").strip()
+        self._update_ui_accorging_to_sso()
+
+    def _update_ui_accorging_to_sso(self):
+        """
+        Updates the GUI if SSO is supported or not, hiding or showing the username/password fields.
+
+        """
+        sso_enabled = is_sso_enabled_on_site(self._url_to_test)
+        if self._use_sso != sso_enabled:
             self._toggle_sso()
 
+    def _site_url_changed(self, text):
+        """
+        Starts a timer to wait until the user stops entering the URL .
+        """
+        self._url_to_test = text.strip()
+        self._url_changed_timer.start(200)
```

Comment:
nitpick: perhaps 200 should be a constant at the top of the file?

Reply:
Done. Now using constant USER_INPUT_DELAY_BEFORE_SSO_CHECK